bin/src/upload.dart

```diff
@@ -27,7 +27,7 @@ class UploadPart extends CommandLinePart {
       });
     }
 
-    var pRoot = new Directory(res["package-root"]);
+    var pRoot = new File(res["packages"]);
```

Comment:
The SDK restriction should stop that in part?  Clients who want to upgrade past dart 1.20 are either going to have to change to use pub get --packages-dir, or will need to update their code to account for .packages anyway?  At least that was my logic when I wrote my version of this.